From fb6d182946df09583c83a01e9552f704b52e7c70 Mon Sep 17 00:00:00 2001
From: Iain Hunter <drhunter95@gmail.com>
Date: Mon, 4 Oct 2021 18:11:30 +0100
Subject: [PATCH 3/3] device tree overlay for inas initial version. ina219
 loaded. ina226 with ultimate address of 0x41 and ads1018 on spi

---
 arch/arm/boot/dts/overlays/Makefile           |  1 +
 .../overlays/ina-evaluation-board-overlay.dts | 67 +++++++++++++++++++
 2 files changed, 68 insertions(+)
 create mode 100644 arch/arm/boot/dts/overlays/ina-evaluation-board-overlay.dts

diff --git a/arch/arm/boot/dts/overlays/Makefile b/arch/arm/boot/dts/overlays/Makefile
index 65e8b3f7c7e8..729155a89352 100644
--- a/arch/arm/boot/dts/overlays/Makefile
+++ b/arch/arm/boot/dts/overlays/Makefile
@@ -92,6 +92,7 @@ dtbo-$(CONFIG_ARCH_BCM2835) += \
 	imx219.dtbo \
 	imx290.dtbo \
 	imx477.dtbo \
+	ina-evaluation-board.dtbo \
 	iqaudio-codec.dtbo \
 	iqaudio-dac.dtbo \
 	iqaudio-dacplus.dtbo \
@@ -186,6 +187,7 @@ dtbo-$(CONFIG_ARCH_BCM2835) += \
 	spi5-2cs.dtbo \
 	spi6-1cs.dtbo \
 	spi6-2cs.dtbo \
+	spidev-disable.dtbo \
 	ssd1306.dtbo \
 	ssd1306-spi.dtbo \
 	ssd1351-spi.dtbo \
diff --git a/arch/arm/boot/dts/overlays/ina-evaluation-board-overlay.dts b/arch/arm/boot/dts/overlays/ina-evaluation-board-overlay.dts
new file mode 100644
index 000000000000..97e5699a1af6
--- /dev/null
+++ b/arch/arm/boot/dts/overlays/ina-evaluation-board-overlay.dts
@@ -0,0 +1,67 @@
+// Definitions for INA219 on I2Cbus
+/dts-v1/;
+/plugin/;
+
+/ {
+	compatible = "brcm,bcm2835";
+
+	fragment@0 {
+		target = <&i2c_arm>;
+		__overlay__ {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			status = "okay";
+			ina219: ina219@40 {
+				compatible = "ti,ina219";
+				reg = <0x40>;
+				/* 0.33ohm fitted */
+				shunt-resistor = <330000>;
+				status = "okay";
+			};
+		};
+	};
+
+	fragment@1 {
+		target = <&i2c_arm>;
+		__overlay__ {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			status = "okay";
+			ina226: ina226@41 {
+				compatible = "ti,ina226";
+				reg = <0x41>;
+				/* 0.33ohm fitted */
+				shunt-resistor = <330000>;
+				status = "okay";
+			};
+		};
+	};
+	
+	fragment@2 {
+		target = <&spi>;
+		__overlay__ {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			status = "okay";
+			iio_adc@1{
+				compatible = "ti,ads1018";
+				reg = <0x1>;
+				#io-channel-cells = <1>;
+				spi-max-frequency = <400000>; 
+				spi-cpha;
+
+				#address-cells = <1>;
+				#size-cells = <0>;
+				// Set up ADS1018 for AIN1
+				channel@5{
+					reg = <5>;      	// for AIN1
+					ti,gain = <2>;  	// 4.096V range
+					ti,datarate = <5>; 	// 2.4kHz sampling
+                };
+			};
+		};
+	};
+};
-- 
2.25.1

